modules/bitgo/test/v2/unit/internal/opengpgUtils.ts

```diff
@@ -61,6 +61,58 @@ describe('OpenGPG Utils Tests', function () {
     });
   });
 
+  describe('verifyShareProof positive case', function () {
+    it('should be able to verify a valid proof', async function () {
+      const uValue = crypto.randomBytes(32).toString('hex');
+      const proof = await openpgpUtils.createShareProof(senderKey.privateKey, uValue);
+      // verify proof
+      const isValid = await openpgpUtils.verifyEdShareProof(senderKey.publicKey, proof, uValue);
+      isValid.should.be.true();
+    });
+  });
+
+  describe('verifyShareProof attack proof', function () {
+    it('should be able to detect sender is an attacker', async function () {
+      const uValue = crypto.randomBytes(32).toString('hex');
+      const proof = await openpgpUtils.createShareProof(otherKey.privateKey, uValue);
+      // verify proof
+      const isValid = await openpgpUtils.verifyEdShareProof(senderKey.publicKey, proof, uValue);
+      isValid.should.be.false();
+    });
+  });
+
+  describe('verifyShareProof attack u value', function () {
+    it('should be able to detect u value is corrupted', async function () {
+      const uValue = crypto.randomBytes(32).toString('hex');
+      const proof = await openpgpUtils.createShareProof(senderKey.privateKey, uValue);
+      const uValueCorrupted = crypto.randomBytes(32).toString('hex');
+      // verify proof
+      const isValid = await openpgpUtils.verifyEdShareProof(senderKey.publicKey, proof, uValueCorrupted);
+      isValid.should.be.false();
+    });
+  });
+
+  describe('verifySharedDataProof and createSharedDataProof test', function () {
+    it('should be able to detect if proof value is corrupted or not', async function () {
+      const sharedData1 = crypto.randomBytes(32).toString('hex');
+      const sharedData2 = crypto.randomBytes(32).toString('hex');
+      const dataToProofArray = [{ name: 's1', value: sharedData1 },
+        { name: 's2', value: sharedData2 }];
+      const dataToProofMap = new Map(
```

Comment:
why not pass in dataToProofArray directly?

Reply:
i had written the verification code to use a map as input, i updated the input now to take in array of key val.